### PR TITLE
NUX & Domains: Privacy Protection is not automatically added

### PR DESCRIPTION
**Pull request by aidvu**
Add privacy protection if new user, and buying a domain

### CODE REVIEW

**Comment by bisko** [comment]
LGTM, works as expected and e2e tests for signup are passing. 👍

File: client/signup/steps/domains/index.jsx [hunk 1]
```diff
@@ -120,6 +120,11 @@ const DomainsStep = React.createClass( {
 					domain: suggestion.domain_name,
 					productSlug: suggestion.product_slug
 				} )
+				: undefined,
+			privacyItem = isPurchasingItem && this.props.domainsWithPlansOnly
```
Comment (klimeryk) [inline client/signup/steps/domains/index.jsx]: I think this needs to be broadened a bit - what about the case when user is not affected by DWPO (an "old" user), but buys a custom domain and selects a Personal (or higher) plan. We should also add the privacy protection for them by default. In other words, when the user is buying a domain from their domain credit, we should check privacy protection for them (regardless of DWPO).

Comment (aidvu) [inline client/signup/steps/domains/index.jsx]: The problem here is that we don't have access to a cart yet, so additional checks are hard. What we might want to do, regardless of the state is to auto-add privacy. 

Opinions? @bisko @meremagee

**Comment by aidvu** [comment]
Abandoning for a back-end fix.

**Pull request closed without merging**